src/compiler/parser.ts

```diff
@@ -1203,7 +1202,7 @@ namespace ts {
                 if (token() !== SyntaxKind.Identifier) {
                     node.originalKeywordKind = token();
                 }
-                node.text = internIdentifier(scanner.getTokenValue());
+                node.text = escapeLeadingUnderscores(internIdentifier(scanner.getTokenValue()));

```

Comment:
This is correct for how we use the `identifiers` map the `internIdentifier` function uses (namely, looking for collisions in generated identifiers).